### ISSUE TITLE
A Massive pop-ups of: "failed to save project" (Pics are attached)

### ISSUE THREAD

**Issue by Gold-b** [stackblitz/bolt.new#2546]
### Describe the bug

1. ![image](https://github.com/user-attachments/assets/0184f8c9-5f00-45d7-b662-ad63aaf2b4ac)

2. ![image](https://github.com/user-attachments/assets/55f4b875-6143-44aa-ae82-68b499ae56da)

### Link to the Bolt URL that caused the error

https://bolt.new/~/sb1-erx4sk

### Steps to reproduce

..

### Expected behavior

1. SAVE THE PROJECT!!!
2. HELP CHAT.. 24/7 FOR YOUR LOYAL CUSTOMERS!!!!

### Screen Recording / Screenshot

Already uploaded

### Platform

```
Browser name = Chrome
Full version = 130.0.0.0
Major version = 130
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 88649836,
  "usedJSHeapSize": 84507148,
  "jsHeapSizeLimit": 4294705152
}
Username = Gold-B
Chat ID = 92cc9e00c839
```

### Additional context

1. SAVE THE PROJECT!!!
2. HELP CHAT.. 24/7 FOR YOUR LOYAL CUSTOMERS!!!!

**Comment by endocytosis** [comment]
Sorry to hear this! Are you able to save now? A number of database improvements have been made recently.

**Comment by ikeithcostello** [comment]
### TLDR: download the project to ensure you don't lose your work
I had a similar experience thrice in the last 24 hours with a project I had previously started.  

I downloaded the project and then went to Stackblitz to continue from wherever it had stopped -- and was shocked to see it had reverted to code from **_hours_** earlier in the day.  I was admittedly less panicky the second and third time. ;)

I have that code in the zip file I downloaded ... is my conversation history lost in this update then?  Do I just push the unzipped project to the stackblitz repo directly or via github or something? What's the easiest way to recover from this?  A **zip file import tool** would be helpful right now.  Thanks for your help!!

**Comment by endocytosis** [comment]
Appreciate the feedback! We are aware of this issue and are tracking is here: #2762.
